UnsolvedSymbolException when package declaration contains comment

I have a file A2.java :
```
package org.demo./*aaaaa*/project1.test;
class A2 {
...
}
```

CompilationUnitContext.java
Line 202:            
```
String qName = this.wrappedNode.getPackageDeclaration().get().getName().toString() + "." + name;
            SymbolReference<ResolvedReferenceTypeDeclaration> ref = typeSolver.tryToSolveType(qName);
            if (ref != null && ref.isSolved()) {
                return SymbolReference.adapt(ref, ResolvedTypeDeclaration.class);
            }
```

The name is "A1“，and the qName is "`org.demo.project1/*aaaaa*/\r\n.test.A1`";
It caused an error of JavaParserTypeSolver.java.

JavaParserTypeSolver.java 

```
    private SymbolReference<ResolvedReferenceTypeDeclaration> tryToSolveTypeUncached(String name) {
        String[] nameElements = name.split("\\.");

        for (int i = nameElements.length; i > 0; i--) {
            StringBuilder filePath = new StringBuilder(srcDir.toAbsolutePath().toString());
            for (int j = 0; j < i; j++) {
                filePath.append(File.separator)
                        .append(nameElements[j]);
            }
            filePath.append(".java");
```

**Test.java**
```
import com.github.javaparser.StaticJavaParser;
import com.github.javaparser.ast.CompilationUnit;
import com.github.javaparser.ast.expr.MethodCallExpr;
import com.github.javaparser.resolution.declarations.ResolvedMethodDeclaration;
import com.github.javaparser.symbolsolver.JavaSymbolSolver;
import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
import com.github.javaparser.symbolsolver.resolution.typesolvers.JavaParserTypeSolver;
import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.List;

public class Test {
    public static void main(String[] args) throws FileNotFoundException {
        TypeSolver[] typeSolverArray = new TypeSolver[2];
        typeSolverArray[0] = new ReflectionTypeSolver(false);
        typeSolverArray[1] = new JavaParserTypeSolver("E:\\workspace-demo\\demo-project1\\src\\main\\java");

        TypeSolver typeSolver = new CombinedTypeSolver(typeSolverArray);

        JavaSymbolSolver javaSymbolSolver = new JavaSymbolSolver(typeSolver);

        StaticJavaParser.getConfiguration().setSymbolResolver(javaSymbolSolver);

        File file = new File("E:\\workspace-demo\\demo-project1\\src\\main\\java\\org\\demo\\project1\\test\\ClassTwo.java");
        CompilationUnit compilationUnit = StaticJavaParser.parse(file);

        List<MethodCallExpr> methodCallList = compilationUnit.findAll(MethodCallExpr.class);
        for (MethodCallExpr mce : methodCallList) {
            ResolvedMethodDeclaration resolvedMethodDeclaration = mce.resolve();

            System.out.println(resolvedMethodDeclaration);
        }

    }
}
```

**ClassOne.java**
```
package org.demo.project1.test;

public class ClassOne {
	public void f1() {
		
	}
}
```

**ClassTwo.java**
```
package org.demo./*aaaaa*/project1.test;
//package org.demo.project1.test;

public class ClassTwo {
	public void test() {
		ClassOne classOne = new ClassOne();
		classOne.f1();
	}
}
```

**Exception**
```
Exception in thread "main" UnsolvedSymbolException{context='classOne.f1()', name='classOne', cause='UnsolvedSymbolException{context='null', name='ClassOne', cause='null'}'}
	at com.github.javaparser.symbolsolver.javaparsermodel.contexts.AbstractJavaParserContext.findTypeDeclarations(AbstractJavaParserContext.java:222)
	at com.github.javaparser.symbolsolver.javaparsermodel.contexts.MethodCallExprContext.solveMethod(MethodCallExprContext.java:166)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solve(JavaParserFacade.java:317)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.solve(JavaParserFacade.java:179)
	at com.github.javaparser.symbolsolver.JavaSymbolSolver.resolveDeclaration(JavaSymbolSolver.java:161)
	at com.github.javaparser.ast.expr.MethodCallExpr.resolve(MethodCallExpr.java:317)
	at Test.main(Test.java:32)
Caused by: UnsolvedSymbolException{context='null', name='ClassOne', cause='null'}
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.convertClassOrInterfaceTypeToUsage(JavaParserFacade.java:713)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.convertToUsage(JavaParserFacade.java:660)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.convert(JavaParserFacade.java:798)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.convert(JavaParserFacade.java:794)
	at com.github.javaparser.symbolsolver.javaparsermodel.declarations.JavaParserVariableDeclaration.getType(JavaParserVariableDeclaration.java:59)
```

--- a/javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/CompilationUnitContext.java
+++ b/javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/CompilationUnitContext.java
@@ -23,8 +23,6 @@ package com.github.javaparser.symbolsolver.javaparsermodel.contexts;
 
 
 
-import static com.github.javaparser.symbolsolver.javaparsermodel.contexts.AbstractJavaParserContext.isQualifiedName;
-
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
@@ -60,6 +58,8 @@ import com.github.javaparser.symbolsolver.resolution.SymbolSolver;
  * @author Federico Tomassetti
  */
 public class CompilationUnitContext extends AbstractJavaParserContext<CompilationUnit> {
+    
+    private static final String DEFAULT_PACKAGE = "java.lang";
 
     ///
     /// Static methods
@@ -199,7 +199,7 @@ public class CompilationUnitContext extends AbstractJavaParserContext<Compilatio
 
         // Look in current package
         if (this.wrappedNode.getPackageDeclaration().isPresent()) {
-            String qName = this.wrappedNode.getPackageDeclaration().get().getName().toString() + "." + name;
+            String qName = this.wrappedNode.getPackageDeclaration().get().getNameAsString() + "." + name;
             SymbolReference<ResolvedReferenceTypeDeclaration> ref = typeSolver.tryToSolveType(qName);
             if (ref != null && ref.isSolved()) {
                 return SymbolReference.adapt(ref, ResolvedTypeDeclaration.class);
@@ -225,7 +225,7 @@ public class CompilationUnitContext extends AbstractJavaParserContext<Compilatio
         }
 
         // Look in the java.lang package
-        SymbolReference<ResolvedReferenceTypeDeclaration> ref = typeSolver.tryToSolveType("java.lang." + name);
+        SymbolReference<ResolvedReferenceTypeDeclaration> ref = typeSolver.tryToSolveType(DEFAULT_PACKAGE+ "." + name);
         if (ref != null && ref.isSolved()) {
             return SymbolReference.adapt(ref, ResolvedTypeDeclaration.class);
         }

--- a/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue3614Test.java
+++ b/javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue3614Test.java
@@ -1,0 +1,36 @@
+package com.github.javaparser.symbolsolver;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.expr.VariableDeclarationExpr;
+import com.github.javaparser.symbolsolver.resolution.AbstractResolutionTest;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+
+public class Issue3614Test extends AbstractResolutionTest {
+
+    @Test
+    void test() {
+        String code = "package java./*aaaaa*/util;\n"
+                + "class Foo {\n"
+                + "  public void test() {\n"
+                + "        ArrayList list = new ArrayList();\n"
+                + "    }"
+                + "}";
+
+        ParserConfiguration configuration = new ParserConfiguration()
+                .setSymbolResolver(new JavaSymbolSolver(new CombinedTypeSolver(new ReflectionTypeSolver())));
+        StaticJavaParser.setConfiguration(configuration);
+
+        CompilationUnit cu = StaticJavaParser.parse(code);
+
+        VariableDeclarationExpr vde = cu.findFirst(VariableDeclarationExpr.class).get();
+        String resolvedType = vde.calculateResolvedType().describe();
+        assertEquals("java.util.ArrayList", resolvedType);
+    }
+}

